File: src/main/kotlin/leetcode/Problem1975.kt
package leetcode

/**
 * https://leetcode.com/problems/maximum-matrix-sum/
 */
class Problem1975 {
    fun maxMatrixSum(matrix: Array<IntArray>): Long {
        TODO()
    }
}

fun main() {
    val prob = Problem1975()
    println(prob.maxMatrixSum(arrayOf(
        intArrayOf(1,-1),
        intArrayOf(-1,1)
    ))) // 4
    println(prob.maxMatrixSum(arrayOf(
        intArrayOf(1,2,3),
        intArrayOf(-1,-2,-3),
        intArrayOf(1,2,3)
    ))) // 16
    println(prob.maxMatrixSum(arrayOf(
        intArrayOf(1,2,3,4),
        intArrayOf(-1,-2,-3,4),
        intArrayOf(1,2,3,4)
    ))) // 38
    println(prob.maxMatrixSum(arrayOf(
        intArrayOf(1,2,3,4),
        intArrayOf(-1,-2,-3,-4),
        intArrayOf(1,2,3,4)
    ))) // 40
}

File: src/main/rust/src/bin/problem1975.rs
pub fn max_matrix_sum(matrix: Vec<Vec<i32>>) -> i64 {
    todo!()
}

fn main() {
    println!("{}", max_matrix_sum(vec![vec![1, -1], vec![-1, 1]])); // 4
    println!(
        "{}",
        max_matrix_sum(vec![vec![1, 2, 3], vec![-1, -2, -3], vec![1, 2, 3]])
    ); // 16
    println!(
        "{}",
        max_matrix_sum(vec![
            vec![1, 2, 3, 4],
            vec![-1, -2, -3, 4],
            vec![1, 2, 3, 4],
            vec![1, 2, 3, 4]
        ])
    ); // 38
    println!(
        "{}",
        max_matrix_sum(vec![
            vec![1, 2, 3, 4],
            vec![-1, -2, -3, -4],
            vec![1, 2, 3, 4],
            vec![1, 2, 3, 4]
        ])
    ); // 40
    println!(
        "{}",
        max_matrix_sum(vec![
            vec![1, 2, 3, 4, 5],
            vec![1, 2, 3, 4, 5],
            vec![1, 2, -3, 4, 5],
            vec![1, 2, 3, 4, 5],
            vec![1, 2, 3, 4, 5],
        ])
    ); // 73
}
